src/io/flutter/logging/FlutterLogTree.java

```diff
@@ -35,47 +34,248 @@
 public class FlutterLogTree extends TreeTable {
 
   private static final SimpleDateFormat TIMESTAMP_FORMAT = new SimpleDateFormat("HH:mm:ss.SSS");
-  private static final DefaultEventCellRenderer DEFAULT_EVENT_CELL_RENDERER = new DefaultEventCellRenderer();
 
-  static class LogTreeModel extends ListTreeTableModelOnColumns {
+  private static class ColumnModel {
+
+    private abstract class EntryCellRenderer extends ColoredTableCellRenderer {
+      @Override
+      protected final void customizeCellRenderer(JTable table,
+                                                 @Nullable Object value,
+                                                 boolean selected,
+                                                 boolean hasFocus,
+                                                 int row,
+                                                 int column) {
+        if (value instanceof FlutterLogEntry) {
+          render((FlutterLogEntry)value);
+        }
+      }
+
+      abstract void render(FlutterLogEntry entry);
+    }
+
+    abstract class Column extends ColumnInfo<DefaultMutableTreeNode, FlutterLogEntry> {
+      private TableCellRenderer renderer;
+
+      boolean show = true;
+
+      Column(String name) {
+        super(name);
+      }
+
+      boolean isVisible() {
+        return show;
+      }
+
+      @Nullable
+      @Override
+      public FlutterLogEntry valueOf(DefaultMutableTreeNode node) {
+        if (node instanceof FlutterEventNode) {
+          return ((FlutterEventNode)node).entry;
+        }
+        return null;
+      }
+
+      @Override
+      public TableCellRenderer getCustomizedRenderer(DefaultMutableTreeNode o, TableCellRenderer renderer) {
+        if (renderer == null) {
+          renderer = getRenderer();
+        }
+        return renderer;
+      }
+
+      abstract TableCellRenderer getRenderer();
```

Comment:
Is it more appropriate to call `createRenderer()`?
It seems that we call it once and cache the value.

Reply:
Sure.  I like that.

Reply:
Fixed. 👍